### PR TITLE
rebase to Docker Distribution v2.6.2

### PR DESCRIPTION
**Pull request by dmage**

### CODE REVIEW

**Comment by dmage** [comment]
flake #14772
/retest extended_conformance_install_update

**Comment by dmage** [comment]
flake #14772
/test extended_conformance_install_update

**Comment by dmage** [comment]
flake #15715

**Comment by dmage** [comment]
flake #8571
/retest

File: vendor/github.com/docker/distribution/vendor/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+The packages in this directory represent dependencies that differ
```
Comment (mfojtik) [inline vendor/github.com/docker/distribution/vendor/README.md]: why we need this commit?

**Comment by mfojtik** [comment]
/approve

we should get the extended registry test working before merging this. also we should consider fork_ami and give the QA something to test before merging as well?

File: vendor/github.com/docker/distribution/vendor/README.md (hunk 1 above)
Comment (dmage) [inline vendor/github.com/docker/distribution/vendor/README.md]: It's a brief description from 41f478a37e7648662945c464511f5a7f285afdd0 about why this vendor/ subdirectory exists and how one can recreate it. I found it useful when I was doing rebase. And yes, it another proof that we constantly loose changes related to distribution.

**Comment by dmage** [comment]
> Submit Queue — PR does not have lgtm label.

It seems the robot isn't going to wait for the green extended test, so I add the `do-not-merge` label.

**Comment by mfojtik** [comment]
@dmage without lgtm label nothing is going to be merged.

**Comment by miminar** [comment]
Fork ami is running at: https://ci.openshift.redhat.com/jenkins/view/All/job/ami_build_origin_int_rhel_fork/43/

**Comment by kargakis** [comment]
> It seems the robot isn't going to wait for the green extended test, so I add the do-not-merge label.

Yes, these extended test are not merge-blocking today. Upstream issue: https://github.com/kubernetes/test-infra/issues/3740

**Comment by dmage** [comment]
flake #16025

**Comment by dmage** [comment]
flake #16023
/retest

**Comment by dmage** [comment]
flake #12273
/test extended_conformance_gce

**Comment by dmage** [comment]
flake #14228

**Pull request merged**